src/module.c

```diff
@@ -6446,6 +6564,19 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
         goto cleanup;
     }
 
+    if (ctx->persistent_client &&
+        c->flags & CLIENT_MULTI &&
+        c->cmd->proc != execCommand &&
+        c->cmd->proc != discardCommand &&
+        c->cmd->proc != multiCommand &&
+        c->cmd->proc != watchCommand &&
+        c->cmd->proc != quitCommand &&
+        c->cmd->proc != resetCommand) {
```

Comment:
let's put that list of commands in some common place so we don't have to repeat it.
we could in theory add a flag to these commands in the json files, but the next best thing is to have a method like isEvalCommand. maybe name it isAbortMultiCommand?

Reply:
ok

Reply:
added a isMultiQueuedCommand(c) to hanlde this case and in server.c